### ISSUE TITLE
[SOGS] Website creation for SOGS report

### ISSUE THREAD

**Issue by holanita** [Green-Software-Foundation/state-of-green-software#35]
Why: to share the SOGS report with the public
What: create interactive pages on the SOGS website to show our findings

Details to be added to [this google document](https://docs.google.com/document/d/1TRzX6PpuD5d1bVFaBnT7yJ1bq6XBWQ8b9ALDA2DkEEU/edit).
See previous discussions [here](https://github.com/Green-Software-Foundation/state-of-green-software/issues/12).

Next steps:
- [x] Ismael to finalize headings **asap** 
- [x] Ismael to share further insights so Osama can review everything and schedule in this task 
- [x] Osama to review and confirm if he needs further input to get started 
- [x] Ismael to update the document as requested by Osama and Asim
- [ ] Osama will likely start the actual work **in December** as per Asim (to be updated if needed)
- [ ] Ismael to finalize sections one by one so Osama can upload the content in batches
- [ ] Osama to adjust content after the consistency review if needed
- [ ] The project should be finalized by the end of Q1 2023 

(this comment will be updated on an ongoing basis as needed)

Staging: https://heartfelt-babka-3b6ea9.netlify.app/

**Comment by holanita** [comment]
@zanete fyi Ismael is currently focusing on writing. Headings have already been confirmed on the linked ticket. Ismael will provide further details in about a week. @Leamsi9 when you are ready please add your input to [this google doc](https://docs.google.com/document/d/1TRzX6PpuD5d1bVFaBnT7yJ1bq6XBWQ8b9ALDA2DkEEU/edit?usp=sharing) for Osama.

**Comment by Leamsi9** [comment]
@zanete, @holanita and @osamajandali 
I have now update the above google doc with data format and examples for all priority sections and a few more. That should be enough to unblock Osama with some lore ipsum. 

Only tricky bit and probably requiring the most dev work is my suggestion for the References section in the google doc. @jawache and Osama need to review for concept fit and logistics.

**Comment by osamajandali** [comment]
@Leamsi9 
Can I have the data and the type of graph for each section?
If it’s unavailable, can I have dummy ones to visualize what kind of data I should put there

**Comment by jawache** [comment]
@Leamsi9 I remember we spoke before the 10th regarding getting a fake section completed for @osamajandali. You can use fake text, fake content, fake reports, fake results but real questions (we are at the stage where we can assign specific questions from the survey to specific sections in this report).

Can you work on this tomorrow morning and share a first draft before the Policy WG call tomorrow.

E.g. this was the last place we left the conversation, ToC headings with subheadings for each of the main types of survey questions

- Knowledge
-- Awareness
(We should at this point be able to say exactly which survey questions will have a chart in which section)
-- Adoption
-- Friction
-- Enablers

I'm expecting:
- Placeholders for charts.
- Placeholders for quotes.
- Where you will expect there to be large chunks of text.
- Where you will be expecting graphics of some form, types of graphics.
- Something a few pages long.

**Comment by Leamsi9** [comment]
Here is a first draft of the Business Case section. It would be good to incorporate the Goldman Sachs survey data here too, but needs more discussion.

https://docs.google.com/document/d/1cgy-38nOnW3L98d7mr4OnInHVxKpR2zd/edit?usp=sharing&ouid=102759123097087113446&rtpof=true&sd=true

**Comment by holanita** [comment]
@Leamsi9 – @jawache gave you very specific details on how to flesh out the content. See above and on Slack as well. We talked about this at the beginning too. Please make sure the content is aligned with what Asim had asked for. As a reminder, this ticket is for Osama so you should only share stuff here once it's good to go. Let's avoid any unnecessary confusion. 

Also, if by Goldman Sachs you mean BCG :) then consider it cancelled please.

**Comment by jawache** [comment]
To be clear we don't need completed draft sections to start the design process, we need something which will start what I imagine will be a lengthy back and forth conversation design reviews etc... I also expect the design to inform the structure of the final report. 

It's unclear _exactly_ what we will need to get a design that we are confident will work but to begin the process I think we'll need: 

1. The outline for all sections so we have an idea of how to design the navigation. I'm not sure the level of detail we need in the outline so put in everything we have right now and if we need more detail we can ask. We had an original doc with lots more detail where I was putting in comments and we were fleshing out the outline, can we get that back? That would be a good starting place.

2. A fake ToC section with fake data, fake text, fake survey insights, fake quotes, fake charts, fake research, fake conclusions. We can start with that and as we progress with the design process there is going to be lots of back and forth, will the design match the report, will the report fit into the design.

This is a web first project, the design of the website should not be an afterthought, the report should be structured around the design we decide on for the website. I want the website structure to be beautiful, navigable, very easy to read, easily sharable, trackable. That will require a lot of effort and time.

**Comment by holanita** [comment]
@Leamsi9 let's discuss in the upcoming call as I can't see it under culture.

**Comment by jawache** [comment]
@osamajandali - based off a call I had with @Leamsi9 and @holanita yesterday I've come up with an approach for the design that I think will work.

https://docs.google.com/document/d/1WKzL20AGOumKn6-P3YFxdv0j7fUEcdz5VKY1LNrYEdo/edit#

And I created a video to walk through my thinking https://www.loom.com/share/133d98744844467286751c625cb74fe4

- Basically the website should be driven by the insights/findings from the report.
- Each insight can be viewed on it's own without reading the rest of the report, this means we will have to repeat content.
- You can go through the insights in any order, read the ones that you are interested in and ignore the rest.

**Comment by holanita** [comment]
cc @zanete so you can incorporate this into the weekly sprint as appropriate

**Comment by osamajandali** [comment]
@jawache Does this mean we will only reply on insights rather than graphs?

**Comment by jawache** [comment]
@osamajandali sorry not sure what you mean? Each insight will have a summary card but when you click into it will be a whole page with lots of things like graphs, text, quotes, links etc...

Happy to jump on a call if you need more context?

**Comment by zanete** [comment]
@holanita - FYI, this is @osamajandali only focus this week and him & @jawache are discussing the needs on a call today.

**Comment by holanita** [comment]
@zanete @osamajandali is this on the right track? let me know if there is anything we should discuss tomorrow

**Comment by osamajandali** [comment]
@jawache 
Here is the prototype : https://www.figma.com/proto/B25vyo3My1SzrksFUe2B4p/GSF-Landing-Pages?page-id=901%3A358&node-id=901%3A1620&viewport=-292%2C1600%2C0.39&scaling=min-zoom&starting-point-node-id=901%3A1620
click on any insight to see the insight page

**Comment by holanita** [comment]
Same here @zanete @osamajandali

**Comment by holanita** [comment]
Hey @osamajandali thanks :) Just three comments at first glance, added them to the figma too.
- can we have at least a few insights above the fold so they can see it right away?
- I know you guys have design guidelines for this, but is there a chance we could make this a tiny bit more colourful?
- Could the top part be a lot more dense with less content? I think that's what Asim had in mind when he drafted [his doc](https://docs.google.com/document/d/1WKzL20AGOumKn6-P3YFxdv0j7fUEcdz5VKY1LNrYEdo/edit#)

**Comment by holanita** [comment]
Hey @osamajandali @zanete I see that you guys have a weekly meeting tomorrow. Pls make sure to cover this task as this is quite important. Also, Namrata is asking when the test site will be available – do you have a timeline for rolling this out?

**Comment by jawache** [comment]
@osamajandali I've (finally) put my comments on the above design :)

**Comment by zanete** [comment]
@holanita 
(CC @osamajandali @jawache )
Unfortunately, we did not have the meeting yesterday. But that shouldn't be in the way of @osamajandali making progress on this task as it is the only focus for him. 
Looking forward to hearing about your plans and progress given the feedback, @osamajandali !

**Comment by osamajandali** [comment]
@zanete  will start working on that 
@jawache thanks for your comments on the design..will work on them

**Comment by zanete** [comment]
Progress delayed by illness, but most things will be finished this week. cc @jawache @holanita

**Comment by osamajandali** [comment]
Here is an [updated version ](https://www.figma.com/proto/B25vyo3My1SzrksFUe2B4p/GSF-Landing-Pages?page-id=901%3A358&node-id=1008%3A358&viewport=-597%2C-798%2C0.6&scaling=min-zoom&starting-point-node-id=1008%3A358) after the comments provided by @jawache and @holanita 

I converted the color palette to the light one so we could have a more colorful layout, and I added an icon for each insight and color for each insight dimension

**Comment by holanita** [comment]
@osamajandali this looks awesome, I really like this new version!

**Comment by zanete** [comment]
@jawache still waiting on your feedback here, blocked until that's provided.

**Comment by zanete** [comment]
Work to be continued as Asim has provided feedback.

**Comment by jawache** [comment]
FYI talking to another researcher this Fri who might start carrying on the work, so this project is probably going to go live again soon.

**Comment by zanete** [comment]
Capturing the status during weekly web call
- very little left to do, soon will be ready for another review

**Comment by zanete** [comment]
Capturing the status during weekly web call
- this was on hold this week, still very little left to do, soon will be ready for another review by the end of the weekend

**Comment by jawache** [comment]
FYI @tamigraph is re-joining the Foundation on Monday so we are back on track with this project. I think @Oleg-Zhymolokhov maybe let's get you onboard with PM'ing this project also since it's almost complete.

**Comment by osamajandali** [comment]
@jawache I have replied for some of your comments on figma and solved other comments from yours
check out the latest version
https://www.figma.com/proto/B25vyo3My1SzrksFUe2B4p/GSF-Landing-Pages?page-id=901%3A358&node-id=1008%3A358&viewport=278%2C33%2C0.12&scaling=min-zoom&starting-point-node-id=1008%3A358&show-proto-sidebar=1

**Comment by jawache** [comment]
Thanks @osamajandali looks great to me!

**Comment by osamajandali** [comment]
@jawache, shall I start executing or wait until the data is ready?
I believe that I need to see data to build the website accordingly

**Comment by zanete** [comment]
Capturing the status on weekly call:
- Will park this until SOGS report is complete

**Comment by jawache** [comment]
I'd prefer if we started with the web dev process, would like this project to be delivered sooner and it gives us more of a chance to review and the report format should be dictated by the website not the other way around, so please go ahead and start.

Get us to a point where we can start trying out how real insights would look as we add them in.

**Comment by osamajandali** [comment]
if the report format should be dictated by the website then yeah I'll start development

**Comment by zanete** [comment]
Capturing the status on weekly web call:
* started on the design implementation, not yet finished, hopes to do by end of week
* then will share for feedback

**Comment by Oleg-Zhymolokhov** [comment]
Hey @osamajandali. Could you share the status of the website development? Are there any blockers/decisions to be made by other parties?

**Comment by osamajandali** [comment]
@Oleg-Zhymolokhov a preview version will be ready by tomorrow.
Let me know if you want to meet to discuss things further

**Comment by osamajandali** [comment]
The main component development is done but I need some dummy data that look like the data that will be provided as it looks empty now.
@Oleg-Zhymolokhov can we have a meeting to show what I have and how we will move forward.
@jawache I would like you to join as well

**Comment by osamajandali** [comment]
Sorry forget to share the preview link: https://heartfelt-babka-3b6ea9.netlify.app/
We need to know what data need to be searchable, because if it is only the title & tags then I will implement client side search but if it will be more than that I need to implement algolia here.

**Comment by Oleg-Zhymolokhov** [comment]
@osamajandali I've shared a folder with you where you can find the data for the website. Please note this is not final and will likely be changed. https://drive.google.com/drive/folders/1VOVi2x7yzvPaSIT9j9wQIXUzSB0yqU7R?usp=sharing

As for the call, I think it makes sense to arrange one when we have some media ready for use. I'll contact you.

**Comment by Oleg-Zhymolokhov** [comment]
@zanete 
1) Yes, every "box" on the website will be an insight.
2) Correct
3) Yes, I believe the call will be more productive if we have some media ready to use.

**Comment by zanete** [comment]
Capturing status of weekly call:
- putting this on hold until all copy is final
- will wait for @Oleg-Zhymolokhov to give a signal when that is the case or if there are any additional tasks to do

**Comment by osamajandali** [comment]
@Oleg-Zhymolokhov I've added some insights as requested [(Link)](https://heartfelt-babka-3b6ea9.netlify.app/). The search and tag search is not yet done until we have an answer to the things that we talked about in our call. 
If you've got the meeting notes from our last call, please put them here just to keep track of the things needed

**Comment by Oleg-Zhymolokhov** [comment]
@osamajandali 
could you add internal ordering based on Insight Number? So that insights would be displayed in ascending order if no other filters are applied.

**Comment by zanete** [comment]
Capturing status in weekly call:
- @Oleg-Zhymolokhov and Osama having meetings about this and other issues, this one is going to be included in the next update

**Comment by zanete** [comment]
Capturing status in weekly call:
- the internal ordering based on Insight Number has been done and will be live in a few mins (with the removal of the search bar)

**Comment by zanete** [comment]
Capturing status in weekly call:
- still needs pushing to the preview

**Comment by osamajandali** [comment]
Published
